project-koku/koku
# Cost 999 add org_unit_id option to AWS materialized views

Fixes [COST-999](https://issues.redhat.com/browse/COST-999)

Changes proposed in this PR:
* add org_unit_id option to AWS view options in report provider map
* update test_forecast to use AWSCostSummaryByAccountP for comparison

Testing Instructions:

_make sure the following variables are set actual values in your env file and not defaults_
```
    AWS_ACCESS_KEY_ID=AWS_ACCESS_KEY
    AWS_SECRET_ACCESS_KEY=AWS_SECRET_KEY
    AWS_RESOURCE_NAME=YOUR_COST_MANAGEMENT_AWS_ARN
```
1. Check endpoints for data
    1. start koku
    2. Hit the following endpoint:
        ```
        [POST] http://localhost:8000/api/cost-management/v1/sources/

        with body:
        {
            "name": "QE AWS",
            "source_type": "AWS",
            "authentication": {
                "credentials": {
                    "role_arn": "arn:aws:iam::##:role/cost-management-qe-1"
                },
                "role_arn": "arn:aws:iam::##:role/cost-management-qe-1"
            },
            "billing_source": {
                "data_source": {
                    "bucket": "cost-management-qe-1"
                },
                "bucket": "cost-management-qe-1"
            }
        }
        ```
    3. Hit the sources endpoint to get the UUID for that source (AWS)
    4. Once you have UUID, hit the crawler with the UUID to start crawling AWS organizations with
        ```
        http://localhost:5042/api/cost-management/v1/crawl_account_hierarchy/?provider_uuid=<uuid_from_step_iii>
        ```
    5. Hit this endpoint to get a list of organizatons
        ```
        http://127.0.0.1:8000/api/cost-management/v1/organizations/aws/
        ```
    6. Hit the following endpoints using any of the org_unit_id returned in step v. Expect to have data in your response.
        ``` 
        # cost
        http://127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?group_by[org_unit_id]=<org_unit_id>

        # storage
        http://127.0.0.1:8000/api/cost-management/v1/reports/aws/storage/?filter[org_unit_id]=<org_unit_id>

        $ instance-type
        http://127.0.0.1:8000/api/cost-management/v1/reports/aws/instance-types/?filter[org_unit_id]=<org_unit_id>
        ```

2. Run unittests with tox
    ```
    tox -e py38 -- api.report.test.aws.test_queries.AWSReportQueryTest
    tox -e py38 -- forecast.test.test_forecast.AWSForecastTest
    ```

    Expect to see this result (for both tests):
   ```
   py38: commands succeeded
   ```  
---
Additional Context:
